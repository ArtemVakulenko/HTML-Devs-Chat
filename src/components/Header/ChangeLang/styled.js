import styled from 'styled-components';

export const StSelect = styled.select`
  color: ${({ colors, theme }) => colors[theme].selectColor};
  padding: 3px;
  option{
    color: ${({ colors, theme }) => colors[theme].selectColor};
  }
`;
