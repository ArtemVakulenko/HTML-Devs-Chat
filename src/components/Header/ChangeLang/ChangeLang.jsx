import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { langs } from 'constants';
import { useTheme } from '../../hooks';
import { StSelect } from './styled';

const ChangeLang = () => {
  const [value, setValue] = useState(localStorage.getItem('lang') || 'en');
  const { colors, theme } = useTheme();
  const { i18n } = useTranslation();
  const { arabian, english, russian } = langs;
  const onChange = (e) => {
    const lang = e.target.value;
    i18n.changeLanguage(lang);
    localStorage.setItem('lang', lang);
    setValue(lang);
  };

  return (
    <StSelect
      theme={theme}
      value={value}
      colors={colors}
      onChange={onChange}
    >
      <option value={english}>{english}</option>
      <option value={russian}>{russian}</option>
      <option value={arabian}>{arabian}</option>
    </StSelect>
  );
};

export default ChangeLang;
