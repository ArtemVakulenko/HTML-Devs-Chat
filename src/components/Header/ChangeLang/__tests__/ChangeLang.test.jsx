import React from 'react';
import ChangeLang from '../ChangeLang';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    i18n: {
      changeLanguage: jest.fn(),
    },
  })),
}));
jest.mock('../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        button: '',
      },
    },
    theme: 'dark',
  })),
}));

describe('ChangeLang', () => {
  it('Should match snapshot', () => {
    const component = global.shallow(<ChangeLang />);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StSelect', () => {
    const component = global.mount(<ChangeLang />);
    expect(component.find('styled__StSelect')).toHaveLength(1);
  });
  it('Should have option', () => {
    const component = global.mount(<ChangeLang />);
    expect(component.find('option')).toHaveLength(3);
  });
  it('Should change lange & change select value', () => {
    const value = 'ru';
    const component = global.mount(<ChangeLang />);
    const select = component.find('styled__StSelect');
    select.simulate('change', { target: { value } });
    expect(select.getDOMNode().value).toBe(value);
  });
});
