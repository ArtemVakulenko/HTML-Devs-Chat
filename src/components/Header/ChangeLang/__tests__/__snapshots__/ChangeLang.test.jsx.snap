// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangeLang Should match snapshot 1`] = `"<select class=\\"styled__StSelect-ldZSMM iNEPZn\\"><option selected=\\"\\" value=\\"en\\">en</option><option value=\\"ru\\">ru</option><option value=\\"ar\\">ar</option></select>"`;
